### ISSUE TITLE
A question on getting garbage in sample.py (Generator)

### ISSUE THREAD

**Issue by hka3rs** [karpathy/nanoGPT#69]
Hi,

I am not an expert in Transformers or DNN, but I followed the steps for training tiny_shakespeare and sampled some tokens after training. I see that it is generating garbage.

Here is what I have used:

1. Running on a CPU in cheap laptop (with character tokenization)

training command:
2. python train.py config/train_shakespeare_char.py --device=cpu --compile=False --eval_iters=20 --log_interval=1 --block_size=64 --batch_size=8 --max_iters=5000

sample command:
3.python sample.py --out_dir=out-shakespeare-char

How do I generate some meaningful shakespeare text? Do I increase max_iters?

train_shakespeare_char.py:

# train a miniature character-level shakespeare model
# good for debugging and playing on macbooks and such

out_dir = 'out-shakespeare-char'
eval_interval = 250 # keep frequent because we'll overfit
eval_iters = 200
log_interval = 10 # don't print too too often

# we expect to overfit on this small dataset, so only save when val improves
always_save_checkpoint = False

wandb_log = False # override via command line if you like
wandb_project = 'shakespeare-char'
wandb_run_name = 'mini-gpt'

dataset = 'shakespeare_char'
batch_size = 64
block_size = 256 # context of up to 128 previous characters

# baby GPT model :)
n_layer = 6
n_head = 6
n_embd = 384
dropout = 0.2

learning_rate = 1e-3 # with baby networks can afford to go a bit higher
max_iters = 5000
lr_decay_iters = 5000 # make equal to max_iters usually
min_lr = 1e-4 # learning_rate / 10 usually
beta2 = 0.99 # make a bit bigger because number of tokens per iter is small

warmup_iters = 100 # not super necessary potentially

# on macbook also add
# device = 'cpu'  # run on cpu only
# compile = False # do not torch compile the model
